main/background.ts

```diff
@@ -54,14 +55,36 @@ ipcMain.on("message", async (event, arg) => {
   event.reply("message", `${arg} World!`);
 });
 
+// TODO make generic function for creating pub+sub endpoints
+// TODO make class for subscription management
+let cachedMetrics: SetMetricsStateActionPayload | null = null;
+let cachedMetricsStr = "";
+// TODO list of webContents
+// let cachedMetricsSubList = [];
+let cachedMetricsIsSubActive = false;
+let cachedMetricsTimeout: NodeJS.Timeout | null = null;
+let cachedMetricsUpdateInProgress = false;
+// TODO move to config
+const currentNodeId = 0;
+
 function metricStringParse(item: PrometheusMetricParser | undefined): number | null {
   if (!item) return null;
   return +item.metrics[0].value;
 }
 
 async function getMetrics(): Promise<SetMetricsStateActionPayload> {
   console.log("DEBUG: getMetrics start");
-  const res = await fetch("http://testnet-3.arweave.net:1984/metrics");
+  const nodeList = config.configHandler.configGetNodes();
+  // temp for development
+  const node = nodeList[currentNodeId] || {
```

Comment:
you'll need find here, this is an array of objects with ids, you could reduce the array to an object of `Record<id, config>`